batchflow/models/eager_torch/layers/conv_block.py

```diff
@@ -0,0 +1,545 @@
+        layer_params.remove('self')
+
+        args = {param: getattr(self, param) if hasattr(self, param) else self.kwargs.get(param, None)
+                for param in layer_params
+                if (hasattr(self, param) or (param in self.kwargs))}
+        return args
+
+    def parse_params(self):
+        """ Create necessary Modules from instance parameters. """
+        layout = self.layout or ''
+        layout = layout.replace(' ', '')
+        if len(layout) == 0:
+            logger.warning('ConvBlock: layout is empty, so there is nothing to do, just returning inputs.')
+            return nn.Sequential([])
+        self.module_layout = ''
+
+
+        layout_dict = {}
+        for letter in layout:
+            letter_group = self.LETTERS_GROUPS[letter]
+            letter_counts = layout_dict.setdefault(letter_group, [-1, 0])
+            letter_counts[1] += 1
+
+        modules, skip_modules, combine_modules = nn.ModuleList(), nn.ModuleList(), nn.ModuleList()
+        layers, residuals = [], []
+
+        for i, letter in enumerate(layout):
+            # Arguments for layer creating; arguments for layer call
+            args = {}
+
+            letter_group = self.LETTERS_GROUPS[letter]
+            layer_name = self.LETTERS_LAYERS[letter]
+            layer_class = self.LAYERS_CLASSES[layer_name]
+            layout_dict[letter_group][0] += 1
+
+            if letter in ['R', 'A', '+', '*', '.']:
+                if len(layers) >= 1:
+                    self.module_layout += '_'
+                    modules.append(nn.Sequential(OrderedDict(layers)).to(self.device))
+                    layers = []
+                self.module_layout += letter
+
+                if letter == 'R':
+                    residuals += [self.inputs]
+
+                    layer_desc = 'Layer {}, letter "{}"'.format(i, letter)
+                    layer = nn.Sequential(OrderedDict([(layer_desc, nn.Sequential().to(self.device))]))
+                    skip_modules.append(layer)
+                elif letter == 'A':
+                    args = self.fill_layer_params(layer_class)
+                    args['mode'] = args.get('mode', 'b')
+                    layer = layer_class(**args).to(self.device)
+                    skip = layer(self.inputs)
+                    residuals += [skip]
+
+                    layer_desc = 'Layer {}, letter "{}"; {} -> {}'.format(i, letter,
+                                                                          get_shape(self.inputs),
+                                                                          get_shape(skip))
+                    layer = nn.Sequential(OrderedDict([(layer_desc, layer)])).to(self.device)
+                    skip_modules.append(layer)
+                elif letter in ['+', '*', '.']:
+                    layer = Combine(op=letter).to(self.device)
+                    shape_before = get_shape(self.inputs)
+                    self.inputs = layer([residuals[-1], self.inputs])
```

Comment:
It is more clear to use `pop` instead of `-1` and  then delete last element. There and elsewhere.